farm/data_handler/processor.py

```diff
@@ -721,6 +715,36 @@ def _create_dataset(self):
         return dataset, tensor_names
 
 
+class TextPairClassificationProcessor(TextClassificationProcessor):
+    """
+    Used to handle text pair classification datasets (e.g. Answer Selection or Natural Inference) that come in
+    tsv format. The columns should be called text, text_b and label.
+    """
+    def __init__(self, **kwargs):
+        super(TextPairClassificationProcessor, self).__init__(**kwargs)
+
+    def file_to_dicts(self, file: str) -> [dict]:
+        column_mapping = {}
+        for task in self.tasks.values():
+            column_mapping[task["label_column_name"]] = task["label_name"]
+
+        dicts = read_tsv_sentence_pair(
+            rename_columns=column_mapping,
+            filename=file,
+            delimiter=self.delimiter,
+            skiprows=self.skiprows,
+            proxies=self.proxies,
+        )
+
+
+        # during tokenization inside TextClassificationProcessor only the "text" field is tokenized
+        # To convert text pairs we combine text and text_b into a tuple, then the tokenizer takes care of
+        # adding separator tokens and correct segment IDs
+        for d in dicts:
+            d["text"] = (d["text"], d["text_b"])
+        return dicts
```

Comment:
Elegant trick! Shall we maybe add a test here to make sure TextPairClassificationProcessor.dataset_from_dicts() always contains correct segment IDs? Just in case huggingface's tokenizer signature changes or we go forward with our own Tokenizer class...might be a tricky bug to identify

Reply:
good idea, added.